charmcraft/commands/store/registry.py

```diff
@@ -440,3 +448,77 @@ def _upload_blob(self, filepath, size, digest):
 
         # finally remove the temp filepath
         os.unlink(filepath)
+
+    def upload_from_local(self, digest: str) -> str:
+        """Upload the image from the local registry."""
+        dockerd = LocalDockerdInterface()
+
+        # validate the image is present locally
+        logger.debug("Checking image is present locally")
+        image_info = dockerd.get_image_info(digest)
+        if image_info is None:
+            return
+        local_image_size = image_info["Size"]
+
+        logger.debug("Getting the image from the local repo; size=%s", local_image_size)
+        response = dockerd.get_streamed_image_content(digest)
+
+        tmp_exported = tempfile.NamedTemporaryFile(mode="wb")
+        extracted_total = 0
+        for chunk in response.iter_content(2 ** 20):
```

Comment:
Any special reason for not using `CHUNK_SIZE` (defined in line 52) here?

Reply:
Good catch! Fixed.